Homework5/Vacations/Vacation.cs

```diff
@@ -0,0 +1,129 @@
+            {
+                if (item.Name != name)
+                {
+                    TimeSpan daysSpan = new TimeSpan(0, 0, 0);
+                    int days = 0;
+
+                    foreach (var emp in SetOfEmployees.Where(n => n.Name == item.Name))
+                    {
+                        daysSpan = emp.LastDay - emp.FirstDay;
+                        days = days + daysSpan.Days;
+
+                    }
+                    daysEach.Add((item.Name, days / SetOfEmployees.Where(n => n.Name == item.Name).Count()));
+                }
+                name = item.Name;
+            }
+            return daysEach.DistinctBy(x => x.Item1).ToList();
+        }
+
+        public List<(int month, int empCount)> GetMonthsWithEmployess()
+        {
+            List<(int, int)> monthsAndNumber = new List<(int, int)>();
+            int count = 0;
+            for (int i = 1; i <= 12; i++)
+            {
+                count = SetOfEmployees.Where(x => x.FirstDay.Month == i || x.LastDay.Month == i).DistinctBy(e => e.Name).ToList().Count();
+                monthsAndNumber.Add((i, count));
+                count = 0;
+            }
+            return monthsAndNumber;
+        }
+
+        public List<(int month, int day)> GetFreeDates()
+        {
+            List<DateTime> allDates = new List<DateTime>();
+            List<DateTime> allOcupiedDates = new List<DateTime>();
+            List<(int, int)> freeMonthAndDay = new List<(int, int)>();
+
+            for (var i = new DateTime(2021, 1, 1); i <= new DateTime(2021, 12, 31); i = i.AddDays(1))
+            {
+                allDates.Add(i);
+            }
+
+            foreach (var emp in SetOfEmployees)
+            {
+                for (var i = emp.FirstDay; i <= emp.LastDay; i = i.AddDays(1))
+                {
+                    allOcupiedDates.Add(i);
+                }
+            }
+
+            allOcupiedDates.Sort();
+
+            foreach (var item in allDates)
+            {
+                int count = 0;
+                foreach (var date in allOcupiedDates)
+                {
+                    if (item == date)
+                    {
+                        count++;
+                    }
+                }
+                if (count == 0)
+                {
+                    freeMonthAndDay.Add((item.Month, item.Day));
+                }
+            }
+
+            return freeMonthAndDay;
+        }
+
+        public bool PersonDatesIntersect()
+        {
+            int count = 0;
+            foreach (var item in SetOfEmployees.GroupBy(x => x.Name).Where(y => y.Count() > 1).ToList())
+            {
+                var firstVacation = SetOfEmployees.Where(x => x.Name == item.Key).ToList()[0];
+                var secondVacation = SetOfEmployees.Where(x => x.Name == item.Key).ToList()[1];
+
+                if (firstVacation.LastDay >= secondVacation.FirstDay && firstVacation.FirstDay <= secondVacation.LastDay)
```

Comment:
consider that black line is a first vacation and red is a second. Pls ensure that condition cover all possible intersections

![image](https://user-images.githubusercontent.com/2743649/152679176-ea125e0b-09d2-4a08-850e-331ecbfd9571.png)